packages/edit-site/src/components/page-pages/index.js

```diff
@@ -200,6 +197,8 @@ export default function PagePages() {
 		[ postStatuses, authors ]
 	);
 
+	const filters = useMemo( () => [ { id: 'search', type: 'search' } ] );
```

Comment:
I'm not sure we're expecting more global filters like that, do we? The only piece of info needed with the current architecture is the key to be used in the query, and I believe we can safely start with a `TextFilter` outside of `Filters` component(render in dataviews.js).

Reply:
I'd prefer not to make it a special case, to be honest. This way, `DataViews` is already future-proof for any other page/endpoint/etc.

Reply:
Though it is a special case, no? It's a global filter. How it could affect any other page etc..?

Reply:
My point is that I prefer to conceptualize it as a filter and do this:

```js
<DataViews
  filters={ filters }
/>
```

rather than making it a special case and do this:

```js
<DataViews
  search={search}
/>
```

The reason is this: what if a different page/endpoint has a global filter that is an `enumeration` or any other type of filter?

The first approach is future-proof: it requires no changes in the future, it already supports any type of filter (`search`, `enumeration`, and whatever we add in the future). The second approach, would require us to make changes in `DataViews` API (new properties, etc.) despite it already supports those other filter types.

Reply:
No strong opinions here, but I still don't get it. We still add the `filters` prop to the API and we're trying to support a use case that we don't have right now. The concept of having a text global filter in a list or in a collection in general is the most common one, by far I think. Maybe some others can chime in, but it's not something that important for now IMO. In my mind the prop would be something like `globalFilter="$key"`.

Reply:
> Maybe some others can chime in, but it's not something that important for now IMO. In my mind the prop would be something like globalFilter="$key".

What you are suggesting would be (note we still have to pass the label as well):

```js
<DataViews
  globalFilter={ {id: "key", label: "Label" } }
/>
```

And what this PR proposes is:

```js
<DataViews
  filters=[
    { id: "key", label: "Label", type: "type" }
  ]
/>
```

It's not that different :)

The only differences I see are:

| | Alternative | Existing |
| --- | --- | --- |
| Property name | `globalFilter` | `filters` |
| Property type | object | array of objects |
| Shape of objects | Only `id` and `label` | The same as field filters. |

I feel strongly about the shape of filters being identical between field filters and global filters: I don't think we should do anything differently.

If it helps unlock this review and merge the PR, I am open to use `globalFilter` as name and use an object instead of an array. We could change that later if needed be.

Reply:
After thinking more about it, while I still believe the global filter is a special case and will be almost a text filter every time, passing filters in general allows a consumer to have more custom filters(if they need to) that are outside the specifics of a field. Let's land this.

Reply:
https://github.com/WordPress/gutenberg/pull/55722